src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1439,6 +1449,12 @@ ISymbol INamedTypeSymbol.AssociatedSymbol
         /// </summary>
         ImmutableArray<string> INamedTypeSymbol.TupleElementNames => this.TupleElementNames;
 
+        /// <summary>
+        /// If this is a tuple type symbol, returns the symbol for its underlying type.
+        /// Otherwise, returns null.
+        /// </summary>
+        INamedTypeSymbol INamedTypeSymbol.TupleUnderlyingType => this.TupleUnderlyingType;
+
```

Comment:
The API is defined in a way, which makes it legal to call it on a non-tuple symbol. In general, asserts do not help external consumers, exceptions do, but I don't think it is worth throwing from this API. 

---
In reply to: [62077342](https://github.com/dotnet/roslyn/pull/11047#discussion_r62077342) [](ancestors = 62077342)